reward/staking_info.go

```diff
@@ -44,21 +44,66 @@ var (
 
 // StakingInfo contains staking information.
 type StakingInfo struct {
-	BlockNum uint64 // Block number where staking information of Council is fetched
+	BlockNum uint64 `json:"blockNum"` // Block number where staking information of Council is fetched
 
 	// Information retrieved from AddressBook smart contract
-	CouncilNodeAddrs    []common.Address // NodeIds of Council
-	CouncilStakingAddrs []common.Address // Address of Staking account which holds staking balance
-	CouncilRewardAddrs  []common.Address // Address of Council account which will get block reward
-	// TODO-klaytn-fund: Provide backward-compatibility to read/write marshalled stakingInfo, and then change KIR/PoC to KCF/KFF
-	KIRAddr common.Address // Address of KIR contract
-	PoCAddr common.Address // Address of PoC contract
+	CouncilNodeAddrs    []common.Address `json:"councilNodeAddrs"`    // NodeIds of Council
+	CouncilStakingAddrs []common.Address `json:"councilStakingAddrs"` // Address of Staking account which holds staking balance
+	CouncilRewardAddrs  []common.Address `json:"councilRewardAddrs"`  // Address of Council account which will get block reward
 
-	UseGini bool
-	Gini    float64 // gini coefficient
+	KCFAddr common.Address `json:"kcfAddr"` // Address of KCF contract
+	KFFAddr common.Address `json:"kffAddr"` // Address of KFF contract
+
+	UseGini bool    `json:"useGini"`
+	Gini    float64 `json:"gini"` // gini coefficient
 
 	// Derived from CouncilStakingAddrs
-	CouncilStakingAmounts []uint64 // Staking amounts of Council
+	CouncilStakingAmounts []uint64 `json:"councilStakingAmounts"` // Staking amounts of Council
+}
+
+// UnmarshalJSON supports json unmarshalling for both oldStakingInfo and StakingInfo
+func (st *StakingInfo) UnmarshalJSON(input []byte) error {
+	type extendedSt struct {
+		BlockNum              uint64           `json:"blockNum"`
+		CouncilNodeAddrs      []common.Address `json:"councilNodeAddrs"`
+		CouncilStakingAddrs   []common.Address `json:"councilStakingAddrs"`
+		CouncilRewardAddrs    []common.Address `json:"councilRewardAddrs"`
+		KCFAddr               common.Address   `json:"kcfAddr"`
+		KFFAddr               common.Address   `json:"kffAddr"`
+		UseGini               bool             `json:"useGini"`
+		Gini                  float64          `json:"gini"`
+		CouncilStakingAmounts []uint64         `json:"councilStakingAmounts"`
```

Comment:
The implicit JSON tag was `"BlockNum"`. Therefore this code cannot read legacy staking info.